### ISSUE TITLE
Add Ability to Create Pre-Releases and Releases

### ISSUE THREAD

**Issue by shockone** [electron-userland/electron-builder#446]
As far as I understand, currently `electron-builder` only allows to create drafts. it would be cool to be able to publish those in order to automate the release process.

**Comment by develar** [comment]
How do trigger release in this case? Tag? I don't like it — https://github.com/electron-userland/electron-builder/issues/340#issuecomment-212274769

**Comment by shockone** [comment]
`build --draft=false --prerelease=true` or similar, if you want to match Github API. The thing is, currently publishing via `electron-builder` _requires_ human interaction: I'd need to open the Github releases page, click "Edit" on the latest draft, and then publish it. It means that there is no way to completely automate the release process, unless I'm missing something.

**Comment by develar** [comment]
> requires human interaction

In your case you need to execute `build --draft=false --prerelease=true` on your development machine, right? i.e. human interaction in any case is required. And it is not good to publish from development machine, it is better to use CI server (Travis).

I am not against, just want to understand your workflow.

**Comment by shockone** [comment]
Not really, the intended setup is to execute this command on Travis every time something is merged into a particular branch. Actually, I already have a workaround using a bash script, if you're interested. https://github.com/shockone/black-screen/blob/master/housekeeping/deploy.sh

**Comment by limoragni** [comment]
I think this would be a very nice addition. In my workflow I would like to mark as pre-release those releases that come from the development branch. As it is now I would have to manually mark each dev release as a pre-release. With this option I would just need to know (on the CI server) which is the branch I'm building and use the flag when it has to be used.

**Comment by develar** [comment]
Proposal:

Env `EP_DRAFT` bool, `EP_PRELEASE` bool.
CLI `draft`, `prerelease`.

Once GitHub will unblock out test account, will be pushed and released.

**Comment by develar** [comment]
@shockone Are new options suitable for you?

**Comment by shockone** [comment]
I haven't tried yet, but they must be, thank you for asking.
